developer-hub/src/polkadot_sdk/substrate.rs

```diff
@@ -0,0 +1,149 @@
+//! # Substrate
+//!
+//! Substrate is a Rust framework for building blockchains in a modular and extensible way. While in
+//! itself un-opinionated, it is the main engine behind the Polkadot ecosystem.
+//!
+//! ## Overview, Philosophy
+//!
+//! Substrate approaches blockchain development with an acknowledgement of a few self-evident
+//! truths:
+//!
+//! 1. Society and technology evolves.
+//! 2. Humans are fallible.
+//!
+//! This, makes the task of designing a correct, safe and long-lasting blockchain system hard.
+//!
+//! Nonetheless, in strive towards achieve this goal, substrate embraces the following:
+//!
+//! 1. Use of **Rust** as a modern, and safe programming language, which limits human error through
+//!    various means, most notably memory and type safety.
+//! 2. Substrate is written from the ground-up with a *generic, modular and extensible* design. This
+//!    ensures that software components can be easily swapped and upgraded. Examples of this is
+//!    multiple consensus mechanisms provided by Substrate, as listed below.
+//! 3. Lastly, the final blockchain system created with the above properties needs to be
+//!    upgradeable. In order to achieve this, Substrate is designed as a meta-protocol, whereby the
+//!    application logic of the blockchain (called "Runtime") is encoded as a WASM blob, and is
+//!    stored in the state. The rest of the system (called "Client") acts as the executor of the
+//!    WASM blob.
+//!
+//! In essence, the meta-protocol of all Substrate based chains is the "Runtime as WASM blob"
+//! accord. This enables the Runtime to become inherently upgradeable, crucially without forks. The
+//! upgrade is merely a matter of the WASM blob being changed in the state, which is, in principle,
+//! same as updating an account's balance. Learn more about this in detail, in
+//! [`crate::reference_docs::wasm_meta_protocol`].
+//!
+//! [`frame`], Substrate's default runtime development library, takes the above safety practices
+//! even further by embracing a declarative programming model whereby correctness is enhanced and
+//! the system is highly configurable through parameterization. Learn more about this in
+//! [`crate::reference_docs::trait_based_programming`].
+//!
+//! ## How to Get Stared
+//!
+//! While most developer develop using [`crate::polkadot_sdk::frame_runtime`], it is worth noting a
+//! few (niche) alternatives:
```

Comment:
I would mention niche alternatives at the end rather than the beginning, not sure how relevant they are for most substrate developers

Reply:
Another alternative I want to add here is @gupnik's version of building a runtime in AssemblyScript.

Reply:
a56d3b110df3e39a0ccaba2db4e5be87bf542a87